django-stubs/core/mail/message.pyi

```diff
@@ -62,42 +69,44 @@ class EmailMessage:
         subject: str = ...,
         body: Optional[str] = ...,
         from_email: Optional[str] = ...,
-        to: Optional[Union[Sequence[str], str]] = ...,
-        bcc: Optional[Union[Sequence[str], str]] = ...,
+        to: Optional[Sequence[str]] = ...,
+        bcc: Optional[Sequence[str]] = ...,
         connection: Optional[Any] = ...,
-        attachments: Optional[Union[List[Tuple[str, Union[str, bytes], str]], List[MIMEText]]] = ...,
+        attachments: Optional[Sequence[Union[MIMEBase, _AttachmentTuple]]] = ...,
         headers: Optional[Dict[str, str]] = ...,
-        cc: Optional[Union[Sequence[str], str]] = ...,
-        reply_to: Optional[Union[List[Optional[str]], str]] = ...,
+        cc: Optional[Sequence[str]] = ...,
+        reply_to: Optional[Sequence[str]] = ...,
     ) -> None: ...
     def get_connection(self, fail_silently: bool = ...) -> Any: ...
     # TODO: when typeshed gets more types for email.Message, move it to MIMEMessage, now it has too many false-positives
     def message(self) -> Any: ...
     def recipients(self) -> List[str]: ...
     def send(self, fail_silently: bool = ...) -> int: ...
-    def attach(
-        self,
-        filename: Optional[Union[MIMEText, str]] = ...,
-        content: Optional[Union[bytes, EmailMessage, SafeMIMEText, str]] = ...,
-        mimetype: Optional[str] = ...,
-    ) -> None: ...
+    @overload
+    def attach(self, filename: MIMEText = ...) -> None: ...
+    @overload
+    def attach(self, filename: None = ..., content: _AttachmentContent = ..., mimetype: str = ...) -> None: ...
+    @overload
+    def attach(self, filename: str = ..., content: _AttachmentContent = ..., mimetype: None = ...) -> None: ...
+    @overload
+    def attach(self, filename: str = ..., content: _AttachmentContent = ..., mimetype: str = ...) -> None: ...
```

Comment:
Could you explain why there's three overloads? Can't it just be one 
```
@overload
def attach(self, filename: Optional[str] = ..., content: _AttachmentContent = ..., mimetype: Optional[str] = ...)
```
?

Reply:
Aside from that, amazing work, thanks=)

Reply:
I think this will pass args like `(None, attachment, None)` while it's not a valid case (according to my understanding of the source code), we should allow only `(None, attachment, mimetype), (filename, attachment, None), (filename, attachment, mimetype)` combinations.

Reply:
Oh, ok. The last two could be merged into one though, with `mimetype: Optional[str] = ...`.